### ISSUE TITLE
Opinion: AI commits too many units away from the frontline, not enough fortification

### ISSUE THREAD

**Issue by ashendashin** [LoneGazebo/Community-Patch-DLL#2814]
_1. Mod version (i.e Date - 4/23):_
11/10

_2. Mod list (if using Vox Populi only, leave blank):_

_3. Error description:_ 
I often find that instead of taking advantage of forests and rivers, the AI would constantly move around and try to attack "poorly" defended areas. I put that in quotes because the token force they usually send isn't near enough to  hold that spot for more than a couple turns. As I usually play aggressively, this makes the early and mid game much easier to exploit. I can't say that I would have the same view as a peaceful civ.

Furthering the point of fortification, I've seen far too many archers sent in front of melee units, preferring to attack the next turn instead of running. Sometimes this works as bait, but I personally find that to be a complete waste of archers.

Overall, the tactical AI is obviously more advanced than before, but this has opened many holes in their defenses.

**Comment by LoneGazebo** [comment]
@ashendashin this is very vague, however @ilteroi may have some thoughts.

**Comment by ashendashin** [comment]
Sorry, I was trying not to go on too long about my grievances on the tactical AI. If needed, I think I can get some good screenshots and saves from my current war. It seems to be a good one for clear examples, though this post was made with multiple games in mind.

**Comment by ilteroi** [comment]
screenshots are always good. also savegames though i might be unable to load them

**Comment by ashendashin** [comment]
Didn't come out as well as I hoped, but I managed to get a couple good screens, along with some extras to point out most of the problems better.

[Example 1.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/589462/Example.1.zip) This one isn't really what I was going for, but the the units around Samsun are worth noting. I wasn't interested in taking Samsun, yet there are strong units being poured through there, not to defend, but to attack. This could be an example of the AI trying to flank me with a small force that just won't do. There may be more units in the fog, but they didn't achieve anything, since future attackers were disembarking one at a time.
![20161114075354_1](https://cloud.githubusercontent.com/assets/21032378/20267131/4d3439fc-aa47-11e6-9d37-a2ff993e7dfb.jpg)

[Example 2.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/589479/Example.2.zip) In this one, I'm just trying to show the clear lack of a proper defense. No one is fortified. Bowmen left out front. Some spearmen near Edirne are trying to be opportunistic when there should be a defensive line until horsemen arrive.
![20161114075354_1](https://cloud.githubusercontent.com/assets/21032378/20267624/9b992ff6-aa49-11e6-8edd-730fbfd00f21.jpg)

This one is just something for something I'd like to say. A few turns before, Mediolanum had no walls. The Ottomans decided to send a swordsman, spearman, 2 composite bowmen, and a catapult to lay siege to that city. Now, I know that the AI can't just change course as soon as failure is evident, but the thing is that they would've failed, walls or no, because their units arrived one at a time. These operations work fine later in the game, but early on this sort of thing cripples armies.
![20161114081847_1](https://cloud.githubusercontent.com/assets/21032378/20268044/7a16e3bc-aa4b-11e6-92f6-483250c67aef.jpg)

[Example 4.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/589544/Example.4.zip) Now for the best one. One of the most fragile frontlines I've seen in VP. Of all the problems I've listed, this is the most prominent one. Ranged units are always found on the front now. In this case, I think the terrain is what really screwed the AI over, but since I only play on the communitas map, this problem arises very often now.
![20161114083252_1](https://cloud.githubusercontent.com/assets/21032378/20268563/af9c6d48-aa4d-11e6-9232-0f7b839d50d9.jpg)

**Comment by LoneGazebo** [comment]
@ilteroi for what it is worth, I have noticed that the AI will sometimes target nonessential cities for close_defense operations. Perhaps the threatened city criteria is not working as intended?

**Comment by HalfEmptyMug** [comment]
Regarding city targeting, here are some screenshots from a game I was running with 10-28

Here Dido is concentrating all of her forces in an attempt to capture Riga (who is my ally through the World Congress)
![](http://i.imgur.com/nPgbrBK.jpg)

However, in doing so she allowed me to march through her empire with little resistance. All of the cities in this screenshot were hers at the beginning of the war, but she's continued the siege even after losing all of her original cities.
![](http://i.imgur.com/cEAj97N.jpg)

I should also mention that when I declared war initially she was about to conquer Austria, but managed to get Maria Theresa to capitulate. This would mean that most of her forces would have been in that area at the time, and I guess valued the city as a higher priority due to proximity?

**Comment by ilteroi** [comment]
@ashendashin : i'm not sure i'll be able to improve this much. i'm also seeing some exposed ranged units, but it's a fine line between too timid and too confident ... also often it's due to limited visibility. additionally, dirty secret, the new tactical AI is only used for offense. defense still uses the old code which is based on rather simplistic heuristics. it's in fact quite hard to arrange a good defense because you have to take into account the opponent's moves as well.

@HalfEmptyMug : that shouldn't be happening ... do you have the OperationalAILog.csv file? when did you declare war on them? before they started the siege or after?

**Comment by ashendashin** [comment]
Damn. Maybe it would do well enough if the AI placed down more citadels? They tend to waste their generals and more citadels could easily make their ranged units proper bait, instead of what there is now. Plus it would increase their overall defenses massively, since they hesitate too much around stealing land.

**Comment by HalfEmptyMug** [comment]
Here is the save and operational log: [Logs and Save - City Targeting.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/592437/Logs.and.Save.-.City.Targeting.zip)
(10-28 No-EUI, FlagPromotions (v.7), IGE (v.39), Quickturns)

I've probably been at war with Dido for around 10 turns, she began the siege right after I declared war since Riga's my Ally. Initially she sent some units to meet my force at the border of Utique, but after I pushed through it seems that the units she's sent since are solely those produced by her cities rather than existing units. The units besieging Riga have been there since the war started. Also, Austria became her vassal right after I declared war, which is why most of her army was away from her capital, but they have remained around Vienna for the most part. I did notice two of Carthage's operations are in fact close_defense operations.
